src/api/client/AccountManagement.ts

```diff
@@ -291,4 +317,35 @@ export class AccountManagement {
    * unlinked to any identity.
    */
   public acceptPrimaryKey: ProcedureMethod<AcceptPrimaryKeyRotationParams, void>;
+
+  /**
+   * Generate an offchain authorization signature with a specified signer
+   *
+   * @param args.signer Signer to be used to generate the off chain auth signature
+   * @param args.target DID of the identity to which signer is targeting the authorization
+   * @param args.expiry date after which the authorization expires
+   */
+  public async generateOffChainAuthSignature(args: {
+    signer: string | Account;
+    target: string | Identity;
+    expiry: Date;
+  }): Promise<`0x${string}`> {
+    const { context } = this;
+
+    const { target, signer, expiry } = args;
+
+    const targetIdentity = asIdentity(target, context);
+
+    const offChainAuthNonce = await targetIdentity.getOffChainAuthorizationNonce();
```

Comment:
If the nonce is on the target, how is this supposed to work with multiple accounts being added in `addSecondaryAccounts`? Shouldn't the user be able to specify the nonce as an option, or is it something like the same nonce is fine since they will get processed in a batch?

Reply:
same nonce is fine for all

Reply:
It should still support a manually specified nonce and default to this value, like we do for transaction options.

I am thinking there could be other auths to sign right after joining an identity and the nonce would let the secondary key do all of its signing at once.